### PR TITLE
Created test for dependency-discovery service

### LINKED ISSUE #2826: Add tests for dependency service
In #2797 we are adding a dependency service, and it currently has no tests.  Let's get initial tests for the HTTP route it exposes (`/projects`) as well as the health check.

### PR DESCRIPTION
**Pull request by TueeNguyen**
## Issue This PR Addresses
Fixes #2826 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
- Add test to `dependency-discovery` service, and healthcheck of the service to `status`

## Steps to test the PR
- `pnpm install`
- `pnpm test src/api/dependency-discovery`

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by JerryHue — DISMISSED** [review]
LGTM. I believe we might want to also "mock" the source of the dependency list, so that way we won't depend on the actual file specifically. We can then write an e2e that uses the whole output of `pnpm list`. We can file a follow up issue for that.

File: src/api/status/src/services.js [hunk 1]
```diff
@@ -9,6 +9,11 @@ const prodUrl = (pathname) => `${prodHost}/${apiVersion}${pathname}`;
 
 // Our current list of microservices
 const services = [
+  // {
```
Comment (humphd) [inline src/api/status/src/services.js:12]: What's happening here?  If this isn't ready to land, don't include it.

File: src/api/dependency-discovery/test/dependency.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const request = require('supertest');
+const dependencyList = require('../src/dependency-list');
+const { app } = require('../src');
+
+jest.mock('process', () => ({
+  cwd: () => 'src/api/dependency-discovery',
```
Comment (humphd) [inline src/api/dependency-discovery/test/dependency.test.js:6]: Agree with @JerryHue , let's mock the file for the dependency list, so we can write tests against it.

File: src/api/dependency-discovery/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const baseConfig = require('../../../jest.config.base');
+
+module.exports = {
+  ...baseConfig,
+  setupFiles: ['<rootDir>/src/api/dependency-discovery/jest.setup.js'],
```
Comment (humphd) [inline src/api/dependency-discovery/jest.config.js:5]: Include the root-level `jest.setup.js` too.

**Review by humphd — CHANGES_REQUESTED** [review]

File: src/api/status/src/services.js (hunk 1 above)
Comment (TueeNguyen) [reply src/api/status/src/services.js:12]: I'll remove it

File: src/api/dependency-discovery/jest.config.js (hunk 3 above)
Comment (TueeNguyen) [reply src/api/dependency-discovery/jest.config.js:5]: Is it this file https://github.com/Seneca-CDOT/telescope/blob/master/test/jest.setup.js?

Comment (humphd) [reply src/api/dependency-discovery/jest.config.js]: Yes.  @rclee91 and @JiaHua-Zou had to deal with something similar in other tests where we didn't do this.  They can guide you.

**Comment by TueeNguyen** [comment]
@JerryHue I changed the `export` in `dependency-list` to `getDependencies` instead of `getDependencies()` so that I could mock the function. Is it ok?

**Comment by humphd** [comment]
```
D:\.pnpm-store\v3
Error: getCacheEntry failed: connect ETIMEDOUT 13.107.42.16:443
```

Where are we at with this test failure?

**Comment by TueeNguyen** [comment]
> ```
> D:\.pnpm-store\v3
> Error: getCacheEntry failed: connect ETIMEDOUT 13.107.42.16:443
> ```
> 
> Where are we at with this test failure?

I reran the checks and it passed, will look into this

**Review by rclee91 — DISMISSED** [review]
I ran the tests locally and didn't get the error either. 
Tests passed

**Review by rclee91 — APPROVED** [review]

**Review by JerryHue — APPROVED** [review]
I ran the test with no issue and it went successful. Nice work!

**Pull request merged**